resallocserver/manager.py

```diff
@@ -516,6 +575,23 @@ def _allocate_more_resources(self, event):
 
             self.allocate(event)
 
+    def _clean_unknown_resources(self, event):
+        with session_scope() as session:
+            dbinfo = session.query(models.Pool).get(self.name)
+            last_cleanup = dbinfo.cleaning_unknown_resources
+
+            delta = datetime.now() - last_cleanup
+            minutes = delta.seconds / 60
+            # We don't want to clean up that often
+            if minutes < 30:
+                return
+
+            worker = CleanUnknownWorker(event, self, res_id=None)
+            worker.run()
```

Comment:
Just a nit, this is run with session locked, and it may take some time ... I could close the session asap, then execute the run() and then open a new session....

Reply:
Good catch, updated.
I previously thought it would not work and complain that we are updating `dbinfo` outside its original session.

Reply:
Yeah, I checked docs and it seems like a detached object - and thouse can be modified in the next session (id is remembered, so modifications generate UPDATE, not INSERT).
The fact that it is detached implies that `session.add()` needs to be called, right?  Because otherwise it is not needed.

Reply:
> The fact that it is detached implies that session.add() needs to be called, right? Because otherwise it is not needed.

IIRC I had it in one session and it was still needed.
But now it is for sure needed because if I remove it, the cleanup is done every `loop`.